src/Plugin/AddPathPlugin.php

```diff
@@ -32,17 +42,39 @@ public function __construct(UriInterface $uri)
         }
 
         $this->uri = $uri;
+
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+        $options = $resolver->resolve($config);
+
+        $this->smart = $options['smart'];
     }
 
     /**
      * {@inheritdoc}
      */
     public function handleRequest(RequestInterface $request, callable $next, callable $first)
     {
-        $request = $request->withUri($request->getUri()
-            ->withPath($this->uri->getPath().$request->getUri()->getPath())
-        );
+        $prepend = $this->uri->getPath();
+        $path = $request->getUri()->getPath();
+
+        if (!($this->smart && substr($path, 0, strlen($prepend)) === $prepend)) {
```

Comment:
can we rewrite this to `if (!$this->smart || !substr(...))`? i feel that is more readable.